File: Problems/Algorithms/1423. Maximum Points You Can Obtain from Cards/max_points_from_cards.go
func maxScore(cardPoints []int, k int) int {
    n := len(cardPoints)
    total := 0
    for i := 0; i < n; i++ {
        total += cardPoints[i]
    }
    if k == n {
        return total
    }

    curr := 0
    for i := 0; i < n-k; i++ {
        curr += cardPoints[i]
    }

    ans := curr
    for i := n-k; i < n; i++ {
        curr += cardPoints[i]
        curr -= cardPoints[i-n+k]
        if curr < ans {
            ans = curr
        }
    }

    return total - ans
}

File: Problems/Algorithms/1423. Maximum Points You Can Obtain from Cards/max_points_from_cards.rs
impl Solution {
    pub fn max_score(card_points: Vec<i32>, k: i32) -> i32 {
        let n = card_points.len();
        let sub_arr = &card_points[..k as usize];
        let mut curr: i32 = sub_arr.iter().sum();
        let mut ans = curr;
        
        for i in 1..=k as usize {
            curr += card_points[n - i] - card_points[k as usize - i];
            ans = ans.max(curr);
        }
        
        ans
    }
}

